benchexec/tools/wasp-c.py

```diff
@@ -0,0 +1,46 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.tools.template
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    This class serves as tool adaptor for WASP-C (https://github.com/wasp-platform/wasp)
+    """
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("wasp-c", subdir="bin")
+
+    def version(self, executable):
+        return self._version_from_tool(executable, "-v")
+
+    def name(self):
+        return "WASP-C"
+
+    def cmdline(self, executable, options, task, rlimits):
+
+        if task.property_file:
+            options = options + ["--property", task.property_file]
+        else:
+            raise benchexec.tools.template.UnsupportedFeatureException(
+                "property file is required"
+            )
+        return [executable] + options + [task.single_input_file]
+
+    def determine_result(self, run):
+        status = result.RESULT_UNKNOWN
```

Comment:
Sorry, I had not noticed this before: I would suggest to return `RESULT_UNKNOWN` only if the tool actually gave up itself, and `RESULT_ERROR` in other cases. This will make it easier to distinguish crashes, timeouts, etc. from unknown results, and will allow BenchExec to add more information about the problem in case the tool terminates abnormally. For example, currently if your tool crashes hard (segfault or so) without the chance to print `WASP crashed`, it would be listed as "unknown" instead of an error.

Also, I see that you return "unknown" in case of timeouts, but BenchExec will overwrite this with "timeout" anyway.

In summary, the intended results would be:
- "timeout" in case of timeout (BenchExec does that automatically, no need to worry)
- "unknown" in case the tool gives up voluntarily
- "done" if everything was fine
- "error" or a different string in case the tool crashes

Reply:
Thank you so much for your help! I've updated the `determine_result` method return values accordingly. :slightly_smiling_face: